src/index.d.ts

```diff
@@ -124,6 +125,14 @@ export interface FastImageProperties {
     testID?: string
 }
 
+export interface PreloadProgressHandler {
+    (loaded: number, total: number): void
+}
+
+export interface PreloadCompletionHandler {
+    (loaded: number, skipped: number): void
```

Comment:
Ideally we should be able to tell the status of individual images with these callbacks. I think SDWebImage might support this now.